### PR TITLE
[FEATURE] v1 59/suite evaluation parameter options

### PR DESCRIPTION
**Pull request by tyler-hoffman**
Resolves V1-59, adding evaluation_parameter_options to expectation suites. Along the way, adds the same property to expectations, and a bit of abstraction around `EvaluationParameterDict` (this can go away once we make `EvaluationParameter` a top-level class, rather than a type aliase).

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/core/expectation_suite.py [hunk 1]
```diff
@@ -150,6 +150,18 @@ def expectation_suite_name(self) -> str:
     def expectation_suite_name(self, value) -> None:
         self.name = value
 
+    @property
+    def evaluation_parameter_options(self) -> tuple[str, ...]:
+        """EvaluationParameter options for this ExpectationSuite.
+
+        Returns:
+            tuple[str, ...]: The keys of the evaluation parameters used in by all Expectations of this suite at runtime.
```
Comment (joshua-stauffer) [inline great_expectations/core/expectation_suite.py]:
```suggestion
            tuple[str, ...]: The keys of the evaluation parameters used by all Expectations of this suite at runtime.
```

File: great_expectations/core/expectation_suite.py [hunk 2]
```diff
@@ -150,6 +150,18 @@ def expectation_suite_name(self) -> str:
     def expectation_suite_name(self, value) -> None:
         self.name = value
 
+    @property
+    def evaluation_parameter_options(self) -> tuple[str, ...]:
```
Comment (joshua-stauffer) [inline great_expectations/core/expectation_suite.py:154]: i'm curious, how does `tuple[str, ...]` differ from `tuple[str]`?

File: great_expectations/expectations/expectation.py [hunk 3]
```diff
@@ -129,9 +133,10 @@ def inner_func(*args: P.args, **kwargs: P.kwargs) -> T:
         configuration: dict | None = kwargs.get("configuration")  # type: ignore[assignment] # could be object?
         if configuration:
             kwargs_dict: dict = configuration.get("kwargs", {})
-            for key, value in kwargs_dict.items():
-                if isinstance(value, dict) and "$PARAMETER" in value.keys():
-                    current_expectation_params.append(value["$PARAMETER"])
+            for value in kwargs_dict.values():
+                if is_evaluation_parameter(value):
+                    key = get_evaluation_parameter_key(value)
+                    current_expectation_params.append(key)
```
Comment (joshua-stauffer) [inline great_expectations/expectations/expectation.py:139]: this is way more legible 👏

File: great_expectations/core/expectation_suite.py (hunk 2 above)
Comment (tyler-hoffman) [reply great_expectations/core/expectation_suite.py:154]: `tuple[str]` is a tuple with exactly one element of type string. The syntax I used is a tuple of any length where all elements are of type string.

File: great_expectations/core/expectation_suite.py (hunk 1 above)
Comment (tyler-hoffman) [reply great_expectations/core/expectation_suite.py]: Fixed!

**Review by joshua-stauffer — APPROVED** [review]
Thanks for the quick turnaround on comments - if ci is happy this looks good to me.

**Pull request merged**